src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx

```diff
@@ -1,14 +1,27 @@
 import React, { useEffect, useState } from 'react';
 import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Input, Alert } from 'reactstrap';
 import AddTeamsAutoComplete from './AddTeamsAutoComplete';
+import { useDispatch } from 'react-redux';
+import { postNewTeam, getAllUserTeams } from '../../../../src/actions/allTeamsAction';
 
 const AddTeamPopup = React.memo(props => {
+  const dispatch = useDispatch();
   const closePopup = () => {
     props.onClose();
   };
 
   const [selectedTeam, onSelectTeam] = useState(undefined);
   const [isValidTeam, onValidation] = useState(true);
+  const [isValidNewTeam, onNewTeamValidation] = useState(true);
+
+  // states and onrs for the new team form
+  const [newTeamName, setNewTeamName] = useState('');
+  const [newTeamIsActive, setNewTeamIsActive] = useState(true);
+
+  const onNameChange = (e) => {
+    setNewTeamName(e.target.value);
+    onNewTeamValidation(true);
```

Comment:
(Line 23) This setter gets called every time the new team name changes i.e each time a letter is typed. This seems costly and redundant, and unnecessary as its not being used for anything. You can probably remove it.